jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpanContext.java

```diff
@@ -107,7 +117,7 @@ public String toString() {
     return contextAsString();
   }
 
-  public static JaegerSpanContext contextFromString(String value)
+  public static JaegerSpanContext contextFromString(String value, JaegerObjectFactory objectFactory)
```

Comment:
To avoid breaking current code (there should be none, but still), keep the old signature, add `@Deprecated` and use the new one with `new JaegerObjectFactory()` (or whatever the default value should be):

```java
/**
* @deprecated see {@link #contextFromString(String, JaegerObjectFactory)}
*/
@Deprecated
public static JaegerSpanContext contextFromString(String value) {
  return contextFromString(value, new JaegerObjectFactory());
}
```

Reply:
@yurishkuro points out these internal classes aren't part of official API and users are very unlikely to be calling `contextFromString` directly.